### ISSUE TITLE
Record date & IP address of EU Cookie Consent

### ISSUE THREAD

**Issue by havenswift-hosting** [cubecart/v6#2034]
As consent now has to be specifically obtained (you can no longer use the privacy policy wording "If you continue to use this site you agree to the use of cookies" etc), I think it is necessary to record consent details for this as well and if the user then logs in during the session, to record that detail in this table.  This table could get pretty large so would also need some way of clearing based on IP address & date and where it can be linked to user data, it would need including on the report and be able to be deleted

**Comment by abrookbanks** [comment]
Also a JS tool to pick the cookies they accept. Fun this...

**Comment by abrookbanks** [comment]
There are third party tools such as https://www.cookiebot.com/en/ too.

**Comment by abrookbanks** [comment]
Introducing all the 3rd party add ons makes this impossible. Every cookies need to be listed and categorised and described. "Essential", "Functional" & "Performance" with the ability to select each one you want.

**Comment by Dirty-Butter** [comment]
I have this wording in my policy - note the link:

The*AddThis and *Google Analytics cookies are non-essential. We do not currently have a way for you to opt out of using these cookies, but you can opt yourself out by following these directions to change your browser cookie settings: www.wikihow.com/Clear-Your-Browser%27s-Cookies

**Comment by abrookbanks** [comment]
![screen shot 2018-05-17 at 15 55 56](https://user-images.githubusercontent.com/4520803/40185566-ce957d9a-59ea-11e8-85e0-c166be066edc.png)

![screen shot 2018-05-17 at 15 49 10](https://user-images.githubusercontent.com/4520803/40185159-ddd134b2-59e9-11e8-80f8-680662c9020d.png)

**Comment by abrookbanks** [comment]
![screen shot 2018-05-17 at 15 57 55](https://user-images.githubusercontent.com/4520803/40185694-1e584fd8-59eb-11e8-8bb2-8719c7185875.png)

**Comment by abrookbanks** [comment]
Log added to admin CP.
![screen shot 2018-06-08 at 13 35 01](https://user-images.githubusercontent.com/4520803/41158425-c7a816d6-6b20-11e8-9279-2eabadc5a5d1.png)

**Comment by Dirty-Butter** [comment]
Well I guess I've wasted your time again. I've played with it all again, and this time the stock2 100% stock is working.

I tried today's commit on my stock2 100% stock install. As soon as I enabled the cookie dialogue and tried to save or clear cache - it crashed. I then commented out the new section in customers.index.inc.php and customers.index.php, cleared cache, and admin loaded as expected.

Then I cleared the browser cache as well and tried it again. This time I was able to save the Extra admin page without issue. But in testing other pages I found that the Customer List still crashes.

DEBUG on Customer List:
[Exception] /home/butter01/public_html/dirtybutterestates.com/XXX/sources/customers.index.inc.php:431 - syntax error, unexpected '}', expecting end of file
